uiplib/gui/gallery.py

```diff
@@ -42,7 +44,12 @@ def set_image(self, imagePath):
                                 label="Blur", showvalue=0)
             self.slider.pack()
         self.image = UipImage(imagePath)
-        self.show_image(self.image)
+        im = Image.open(imagePath)
+        try:
+            im.load()
```

Comment:
not sure myself, yes the UIP image class should be used but even then i'm not sure we should check here